src/pages/home/report/ReportActionCompose.js

```diff
@@ -66,18 +72,23 @@ class ReportActionCompose extends React.Component {
         this.setIsFocused = this.setIsFocused.bind(this);
         this.focus = this.focus.bind(this);
         this.comment = props.comment;
+        this.shouldFocusInputOnScreenFocus = canFocusInputOnScreenFocus();
 
         this.state = {
-            isFocused: true,
+            isFocused: this.shouldFocusInputOnScreenFocus,
             textInputShouldClear: false,
             isCommentEmpty: props.comment.length === 0,
             isMenuVisible: false,
         };
     }
 
     componentDidUpdate(prevProps) {
-        // When any modal goes from visible to hidden, bring focus to the compose field
-        if (prevProps.modal.isVisible && !this.props.modal.isVisible) {
+        // We want to focus or refocus the input when a modal has been closed and the underlying screen is focused.
+        // We avoid doing this on native platforms since the software keyboard popping
+        // open creates a jarring and broken UX.
+        if (this.shouldFocusInputOnScreenFocus && this.props.isFocused
+            && prevProps.modal.isVisible && !this.props.modal.isVisible) {
+            this.setIsFocused(true);
```

Comment:
`setIsFocused(true)` would be called when the field is focused anyway. 
Though if there is a legit reason to call it here, the call can just be moved inside the `focus` method

```js
this.setIsFocused(true);
this.focus();
```

vs

```js
this.focus();
```

Reply:
Nice thoughts. Yeah Cancellation of Interaction is good to have. 

But about autofocus, I think there are some PR on react native that are handling animation and focus issue. But I agree lets just move the interaction based focus to TextInputFocusable. I didn't do this here as that covers bigger scope then the current issue. So its better to first analyze the effect of these changes and consider them.